wit/basic-consumer.wit

```diff
@@ -0,0 +1,7 @@
+/// A basic consumer, commonly used for load-balancing.
+default interface basic-consumer {
+	use pkg.types.{message, error, receiver}
+
+	/// `receive-message` satisfied pull-based messaging requirements
+	receive-message(r: receiver) -> result<message, error>
```

Comment:
We need settlement (complete, abandon, deadletter or accept, release, reject) as first-class constructs for all queue-based interactions. The decision on how to settle a message is often dependent on the application outcome.

In queues, all messages are settled singly. MQTT also requires accept/reject (positive/negative ACK) for QoS 1/2.

You could put a `receive-context` on the message that has appropriate settlement methods. That would then also work with the reactive API style below.